CIP-0021/CIP-0021.md

```diff
@@ -0,0 +1,139 @@
+
+**Unsupported entries**
+
+The transaction body entry `6 : update` must not be included.
+
+**Integers**
+
+While the Cardano CDDL specification usually does not limit the size of integers, HW wallets only support `int64` for signed integers and `uint64` for unsigned integers. Any integer value must fit in the appropriate type.
+
+**Numbers of transaction elements**
+
+The numbers of inputs, outputs, certificates, withdrawals, tokens in an asset group, and other similar transaction elements must not exceed `UINT16_MAX`, i.e. 65535.
+
+**Optional empty lists and maps**
+
+Unless mentioned otherwise in this CIP, optional empty lists and maps must not be included as part of the transaction body or its elements.
+
+**Outputs**
+
+Outputs containing no multi-asset tokens must be serialized as a simple tuple, i.e. `[address, coin]` instead of `[address, [coin, {}]]`.
+
+**Multiassets**
+
+Since multiassets (`policy_id` and `asset_name`) are represented as maps, both need to be sorted in accordance with the specified canonical CBOR format. Also, an output or the mint field must not contain duplicate `policy_id`s and a policy must not contain duplicate `asset_name`s.
+
+**Certificates**
+
+Certificates of type `genesis_key_delegation` and `move_instantaneous_rewards_cert` are not supported and must not be included.
+
+If a transaction contains a pool registration certificate, then it must not contain:
+
+- any other certificate;
+- any withdrawal;
+- mint entry.
+
+It is allowed to arbitrarily combine other supported certificate types, but all the certificates included in a transaction must have the same type of stake credential i.e. either 0 - `addr_keyhash` or 1 - `scripthash`. The stake credential type must also be consistent with the type used for withdrawals.
+
+**Withdrawals**
+
+Since withdrawals are represented as a map of reward accounts, withdrawals also need to be sorted in accordance with the specified canonical CBOR format. A transaction must not contain duplicate withdrawals. All withdrawals included in a transaction must have the same type of stake credential i.e. either 0 - `addr_keyhash` or 1 - `scripthash`. The stake credential type must also be consistent with the type used for certificates.
+
+**Auxiliary data**
+
+HW wallets do not serialize auxiliary data because of their complex structure. They only include the given auxiliary data hash in the transaction body. The only exception is Catalyst voting registration because it requires a signature computed by the HW wallet.
+
+In this exceptional case, auxiliary data must be encoded in their "tuple" format:
+
+```
+[ transaction_metadata: { * transaction_metadatum_label => transaction_metadatum }, auxiliary_scripts: [ * native_script ]]
+```
+
+The `auxiliary_scripts` must be an array of length 0.
```

Comment:
Is that something which is in the making, or there's no plan to support auxiliary_scripts ever? That may be quite limiting for Plutus.

Reply:
If `auxiliary_scripts` support is needed on HW wallets it should be added. Currently it's not supported and so it is a restriction on the transactions.